src/lib/eZ/RichText/Resources/schemas/docbook/ezpublish.rng

```diff
@@ -117,6 +118,16 @@
         </choice>
       </interleave>
     </define>
+    <define name="db.xlink.href.attribute">
+      <attribute name="xlink:href">
+        <a:documentation>Identifies a link target with a URI</a:documentation>
+        <choice>
+          <data type="string">
+            <param name="pattern">(https?://|mailto:|ezcontent://|ezlocation://|ezremote://|ezurl://|tel:|/|#).+</param>
```

Comment:
The original fix for [ibexa-sa-2024-005-persistent-xss-in-richtext](https://developers.ibexa.co/security-advisories/ibexa-sa-2024-005-persistent-xss-in-richtext) also checks for the existence of `&lt;`, `&gt;` and `'&quot;`

In order to check for this in the reg-ex, I need to use [Negative Lookahead](https://www.regular-expressions.info/lookaround.html), but adding `(?!foobar)` in the regex is not accepted by the schema validator, ie:
`<param name="pattern">(?!.*foobar)(https?://|mailto:|ezcontent://|ezlocation://|ezremote://|ezurl://|tel:|/|#).+</param>-`

It returns erros like:
```
Validation of XML content failed: Fatal error in 0:0: failed to compile: xmlFAParseAtom: expecting ')'
Validation of XML content failed: Fatal error in 0:0: failed to compile: xmlFAParseRegExp: extra characters
```

regex engine do not support lookahead?

Reply:
The check for  `&lt;`, `&gt;` and `'&quot;` is uneeded as urls are correctly URL encoded when sent to front-end anyway